transcendence-app/src/game/game.module.ts

```diff
@@ -2,12 +2,33 @@ import { Global, Module } from '@nestjs/common';
 import { DbModule } from '../shared/db/db.module';
 import { GameGateway } from './game.gateway';
 import { AuthorizationModule } from '../authorization/authorization.module';
+import { GameQueueGateway } from './game.queue.gateway';
+import { GameQueueService } from './game.queue.service';
+import {
+  ChallengesPending,
+  GamesOngoing,
+} from './infrastructure/db/memoryModels';
+import { IGamesOngoingRepository } from './infrastructure/db/gamesongoing.repository';
+import { IChallengesPendingRepository } from './infrastructure/db/challengespending.repository';
+import { GameController } from './game.controller';
 
 @Global()
 @Module({
   imports: [DbModule, AuthorizationModule],
-  providers: [GameGateway],
-  // controllers: [ChatController],
+  providers: [
+    GameGateway,
+    GameQueueGateway,
+    GameQueueService,
+    {
+      provide: IChallengesPendingRepository,
+      useClass: ChallengesPending,
```

Comment:
I prefer to state the most encompassing name in the first place, as if I was using namespaces, even it's not gramatically correct english. After all, it's not a sentence: it's the name of a class/type.

Reply:
To explain how come I find this weird incorrect english more programatically accurate I'll give an example.

Imagine we have four types (that we actually don't because not all of them make actual sense in our application):

Pending games,
Ongoing games,
pending challenges,
ongoing challenges.

these four are, in the end, four files in a folder. According to your convention they would be (for example, ignoring the `.repository` or whatever part), when alphabetically sorted:

```
ongoing-challenges.ts
ongoing-games.ts
pending-challenges.ts
pending-games.ts
```

but I'd rather have them sorted as:

```
ongoing-challenges.ts
pending-challenges.ts
ongoing-games.ts
pending-games.ts
```
How do you group the sorted files by larger namespace?
Well, you put the larger namespace in the name in front. 
```
challenges-ongoing.ts
challenges-pending.ts
games-ongoing.ts
games-pending.ts
```